### PR TITLE
Add staging and sandbox environments 

### PR DESCRIPTION
**Pull request by otterbotter**
Related ansible changes PR: https://github.com/OpenUpSA/ansible-config/pull/53

This PR:
- Adds build and environment context settings to `netlify.toml` to allow different environment variables to be set in different contexts/environments
- Adds new environment variables: `DEFAULT_API_URL`, `ENVIRONMENT` and `PUSHPAD_PROJECT_ID`
- Changes how pushpad is initialised: doesn't if no project id set, and reads project id from environment variable
- Sets default API URL to `DEFAULT_API_URL`, if not set it defaults to production
- Sentry will now initialise not only in production, but whenever `SENTRY_DSN` is set. So for `production`, `sandbox` and `staging` we set `SENTRY_DSN`, for other environments we set it to an empty string.
- Adds `<meta name="robots" content="noindex">` during build process if not production, to prevent search engines from indexing non-prod sites


💀 Warning:
The file-based configuration (`netlify.toml`) will only take effect when this PR is merged into master. Only after we merge will we know whether or not it works correctly. If something is wrong, it'll likely break the production build and we'll have to fix it in the configuration file.

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -1,4 +1,56 @@
...
+[context.staging.environment] # 'staging' is a branch
+    GOOGLE_TAG_MANAGER_ID = ""
+    SENTRY_DSN = "https://a5d3ce913bdd4eb594da36387b286d83@o242378.ingest.sentry.io/5496000"
+    SENTRY_PERF_SAMPLE_RATE = "0"
+    DEFAULT_API_URL = "staging.muni-portal-backend.openup.org.za"
+    ENVIRONMENT = "staging"
+    PUSHPAD_PROJECT_ID = "7570"
+
+# Specific branch context: all deploys from this specific branch will inherit these settings.
+[context.sandbox.environment] # 'sandbox' is a branch
```
Comment (jbothma) [inline netlify.toml:32]: could the sandbox environment be configured to deploy from the master branch, just like production?

File: src/js/webflow/import.js [hunk 2]
```diff
@@ -9,14 +9,18 @@ exports.transform = function (window, $) {
...
-    (function (p, u, s, h, x) { p.pushpad = p.pushpad || function () { (p.pushpad.q = p.pushpad.q || []).push(arguments) }; h = u.getElementsByTagName('head')[0]; x = u.createElement('script'); x.async = 1; x.src = s; h.appendChild(x); })(window, document, 'https://pushpad.xyz/pushpad.js');
-
-    pushpad('init', '7550', {serviceWorkerPath: null});
-    pushpad('widget', {
-      promptTitle: 'Subscribe to Notifications',
-      promptMessage: 'Subscribe to receive instant notifications of service disruptions and other important information.',
-      promptButtonColor: "#0094ff",
-    });
+      const pushpadProjectId = \`${process.env.PUSHPAD_PROJECT_ID}\`;
+      if (!pushpadProjectId !== '') {
```
Comment (jbothma) [inline src/js/webflow/import.js:13]:
```suggestion
      if (pushpadProjectId !== '') {
```

Comment (jbothma) [reply src/js/webflow/import.js:13]: I don't think you want double `not`

Will it be empty string, null or undefined?

File: netlify.toml (hunk 1 above)
Comment (otterbotter) [reply netlify.toml:32]: Not as far as I can see. Initially that was the plan but it seems to me like the branch has to exist for Netlify to be able to make a branch deploy.

File: src/js/webflow/import.js (hunk 2 above)
Comment (otterbotter) [reply src/js/webflow/import.js]: Thanks for catching that

It'll be an empty string, since it is initialised as a template literal

**Pull request merged**